### ISSUE TITLE
Why is Swift's `isMultiple(of:)` function so slow?

### ISSUE THREAD

**Issue by abzr1** [swiftlang/swift#75826]
### Description

I have the following two code samples:

```swift
func isPrime(_ n: Int) -> Bool {
    if n < 2 {
        return false
    }

    var i = 2

    while i * i <= n {
        if n.isMultiple(of: i) { // Notice this line
            return false
        } else {
            i += 1
        }
    }

    return true
}

var sumOfPrimes = 0

for i in 0...1_000_000 {
    if isPrime(i) {
        sumOfPrimes += i
    }
}

print(sumOfPrimes)
```

```swift
func isPrime(_ n: Int) -> Bool {
    if n < 2 {
        return false
    }

    var i = 2

    while i * i <= n {
        if n % i == 0 { // Notice this line
            return false
        } else {
            i += 1
        }
    }

    return true
}

var sumOfPrimes = 0

for i in 0...1_000_000 {
    if isPrime(i) {
        sumOfPrimes += i
    }
}

print(sumOfPrimes)
```

Those two lines *should* be doing exactly the same thing, but when I time them, I get, respectively:

```shell
$ time swift x.swift
37550402023
swift x.swift  2.84s user 0.02s system 98% cpu 2.887 total
```

```shell
$ time swift x.swift
37550402023
swift x.swift  0.40s user 0.02s system 95% cpu 0.437 total
```

The results are consistent. Why is Swift's `isMultiple(of:)` function so slow?

### Environment

Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-darwin23.5.0

**Comment by thomasjconnolly** [comment]
isMultiple(of other: Int) is labeled as @inlinable, so it appears to be a function which can be inlined, but is not inlined by default. When you changed the code, you are inlining the function. So it appears what you are testing is the speed in inlining versus not inlining. 

When I ran the example you provided, I got 2.30 and 0.36 seconds.

Running this with command line option "-O" to add optimization, both examples execute for me in 0.15-0.18 seconds.
time swift -O x.swift
